typescript/sdk/src/hook/types.ts

```diff
@@ -1,22 +1,43 @@
+import { BigNumber } from 'ethers';
+
 import type { Address } from '@hyperlane-xyz/utils';
 
+import type { MultisigIsmConfig } from '../ism/types';
 import { ChainName } from '../types';
 
-export enum HookContractType {
+export enum InterceptorType {
   HOOK = 'hook',
   ISM = 'ism',
 }
 
-export type MessageHookConfig = {
-  hookContractType: HookContractType.HOOK;
+export type OpStackHookConfig = {
+  type: InterceptorType.HOOK;
   nativeBridge: Address;
   remoteIsm?: Address;
+  destinationDomain: BigNumber;
   destination: ChainName;
 };
 
+export type MerkleRootHookConfig = {
+  type: InterceptorType.HOOK;
+};
+
+export type MerkleRootInterceptorConfig = {
+  hook: MerkleRootHookConfig;
+  ism: MultisigIsmConfig;
+};
+
+export type OpStackInterceptorConfig = {
+  hook: OpStackHookConfig;
+  ism: NoMetadataIsmConfig;
```

Comment:
I would expect "no metadata ISM" to not require config

Reply:
That was OP specific